src/ingredient_cache.rs

```diff
@@ -4,26 +4,33 @@ use std::num::NonZeroU32;
 
 use crate::nonce::Nonce;
 use crate::plumbing::Ingredient;
-use crate::sync::atomic::{AtomicU32, AtomicU64, Ordering};
+use crate::sync::atomic::{AtomicU64, Ordering};
 use crate::zalsa::{StorageNonce, Zalsa};
 use crate::IngredientIndex;
 
+#[cfg(not(feature = "inventory"))]
+/// With manual registration, ingredient indices can vary across databases,
+/// so the `GlobalIngredientCache` optimization does not apply.
+pub use IngredientCache as GlobalIngredientCache;
+
 /// Caches an ingredient index.
 ///
 /// Unlike [`IngredientCache`], this is not restricted to a specific database.
-/// Note that ingredients are statically registered with `inventory`, so their
+/// Note that all ingredients are statically registered with `inventory`, so their
 /// indices should be stable across any databases.
 ///
 /// If ingredient initialization is database dependent, e.g. for registering
 /// view casters, [`IngredientCache`] should be used instead.
+#[cfg(feature = "inventory")]
 pub struct GlobalIngredientCache<I>
 where
     I: Ingredient,
 {
-    ingredient_index: AtomicU32,
+    ingredient_index: std::sync::atomic::AtomicU32,
```

Comment:
What's the reason for using the `std::sync::atomic` here over our version in `crate::sync`?